specs/mgt-search-suggestion.md

```diff
@@ -0,0 +1,161 @@
+<img src="./images/mgt-search-suggestion.png" width=400/>
+
+## Supported functionality
+
+| Feature | Priority | Notes |
+| ------- | -------- | ----- |
+| Retreieve text/people/file/ from Microsoft Graph endpoint based on the query string | P0 | |
+| Display the text of the matched query and display the name of the file/people| P0| |
+| Display the headImg or create a head icon for suggested people| P0| |
+| Display an icon indicating if it's a folder or file and the file type| P0 | Icons needed include generic folder icon, .docx, .pptx, .xlsx, and generic file icon for other file types |
+| Display relevant details of the file | P0 | Developer should be able to configure what details are being rendered |
+| Provide basic default key down action and mouse action | P0 | when mouse move to an item, the backgroud/cursor should change. when arrow up/down, one item should be selected|
+| Provide callback function to get data for supporting users customization actions | P0 | when user click one item, should provide a way for getting the suggested data and support customize actions. when enter key pressed, should provide a way for getting the focused data and support customize actions|
+
+## Templates
+
+ `mgt-search-suggestion` supports several [templates](../customize-components/templates.md) that you can use to replace certain parts of the component. To specify a template, include a `<template>` element inside a component and set the `data-type` value to one of the following.
+
+| Data type | Data context | Description |
+| --- | --- | --- |
+| default | null: no data | The template used to override the rendering of the entire component.
+| loading | null: no data | The template used to render the state of picker while request to graph is being made. |
+| error | null: no data | The template used if user search returns no users. |
+| no-data | null: no data | An alternative template used if user search returns no users. |
+| search-suggestion-input | null: no data | The template to render search input box. |
+| search-suggestion-label-people | null: no data | The template to render people entity label. |
+| search-suggestion-label-text | null: no data | The template to render text entity label. |
+| search-suggestion-label-file | null: no data | The template to render file entity label. |
+| search-suggestion-people | suggestionPeople[]: The people suggestion details list | The template to render people entity. |
+| search-suggestion-text | suggestionText[]: The text suggestion details list | The template to render text entity. |
+| search-suggestion-file | suggestionFile[]: The file suggestion details list | The template to render file entity. |
```

Comment:
Few suggestions to simplify template names
- Header seems more appropriate than label
- Remove `search` from data-type name
- simplify data context name for better developer experience

Also, I propose the `sugested-*` templates to be a template for a single entity and not a list

```suggestion
| input | null: no data | The template to render search input box. |
| suggested-people-header | null: no data | The template to render people entity label. |
| suggested-queries-header | null: no data | The template to render text entity label. |
| suggested-file-header | null: no data | The template to render file entity label. |
| suggested-person | person: A suggested person details | The template to render suggested person. |
| suggested-query | query: A string query suggestion | The template to render suggested query. |
| suggested-file | file: A suggested file details | The template to render suggested file. |
```

Reply:
yeah, let's use header. and I will try to template a single entity.

Reply:
Resolved current now